compiler/src/iree/compiler/Reducer/Framework/Delta.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Reducer/Framework/Delta.h"
+
+using namespace mlir;
+using namespace mlir::iree_compiler;
+
+FailureOr<WorkItem> Delta::checkChunk(Chunk maybeUninterestingChunk,
+                                      DeltaFunc deltaFunc,
+                                      ArrayRef<Chunk> maybeInterestingChunks,
+                                      DenseSet<Chunk> &uninterestingChunks) {
+  SmallVector<Chunk> currentChunks;
+  copy_if(maybeInterestingChunks, std::back_inserter(currentChunks),
+          [&](Chunk chunk) {
+            return chunk != maybeUninterestingChunk &&
+                   !uninterestingChunks.count(chunk);
+          });
+
+  debugOs << "Checking chunk: \n";
+  maybeUninterestingChunk.dump();
```

Comment:
Stray dumps here and throughout? I would have expected that if keeping these, you would just be emitting to debugOs (which may or may not be stderr).